src/execution_engine.rs

```diff
@@ -470,7 +476,30 @@ mod private {
 macro_rules! impl_unsafe_fn {
     ($( $param:ident ),*) => {
         impl<Output, $( $param ),*> private::Sealed for unsafe extern "C" fn($( $param ),*) -> Output {}
+
         impl<Output, $( $param ),*> UnsafeFunctionPointer for unsafe extern "C" fn($( $param ),*) -> Output {}
+
+        impl<Output, $( $param ),*> Symbol<unsafe extern "C" fn($( $param ),*) -> Output> {
+            /// This method allows to call the underlying function while making
+            /// sure that the backing storage is not dropped too early and
+            /// preserves the `unsafe` marker for any calls.
+            #[allow(non_snake_case)]
```

Comment:
Why is `non_snake_case` needed here? `call` is snake case?

Reply:
This is for parameter names (otherwise macro would have to be passed `a: A, b: B, ...` and I don't think it's worth it.